src/conditionals/simple_programs_with_decision_points/hard/q4/MainTest.java

```diff
@@ -0,0 +1,42 @@
+package conditionals.simple_programs_with_decision_points.hard.q4;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+public class MainTest extends BaseTest {
+    // Parsons with distractors
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "true";
+
+        return new Clause[]{
+                new StringLiteral("Is the door open or closed right now\\? \\(Enter true for open and false for closed\\)"),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        DoorBoy.main(new String[0]);
+    }
+
+    @Test
+    public void testOpen() throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program should have closed the door, it was already open.";
```

Comment:
The error messages in this file would be weird to read from a student's perspective since they won't (once we fix it) know what the input was that caused the test to fail.

Perhaps something more like:
"Your program attempted to (open/close) an already (opened/closed) door."